hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -655,4 +786,113 @@ private static Schema resolveNullableSchema(Schema schema) {
 
     return nonNullType;
   }
+
+  // TODO elaborate
+  public static class ConvertingGenericData extends GenericData {
+
+    private static final DecimalConversion DECIMAL_CONVERSION = new DecimalConversion();
+    private static final Conversions.UUIDConversion UUID_CONVERSION = new Conversions.UUIDConversion();
+    private static final TimeConversions.DateConversion DATE_CONVERSION = new TimeConversions.DateConversion();
+    private static final TimeConversions.TimeMicrosConversion TIME_MICROS_CONVERSION = new TimeConversions.TimeMicrosConversion();
+    private static final TimeConversions.TimestampMicrosConversion TIMESTAMP_MICROS_CONVERSION = new TimeConversions.TimestampMicrosConversion();
+
+    // NOTE: Those are not supported in Avro 1.8.2
+    // TODO re-enable upon upgrading to 1.10
+    // private static final TimeConversions.TimestampMillisConversion TIMESTAMP_MILLIS_CONVERSION = new TimeConversions.TimestampMillisConversion();
+    // private static final TimeConversions.TimeMillisConversion TIME_MILLIS_CONVERSION = new TimeConversions.TimeMillisConversion();
+    // private static final TimeConversions.LocalTimestampMillisConversion LOCAL_TIMESTAMP_MILLIS_CONVERSION = new TimeConversions.LocalTimestampMillisConversion();
+    // private static final TimeConversions.LocalTimestampMicrosConversion LOCAL_TIMESTAMP_MICROS_CONVERSION = new TimeConversions.LocalTimestampMicrosConversion();
+
+    public static final GenericData INSTANCE = new ConvertingGenericData();
+
+    private ConvertingGenericData() {
+      addLogicalTypeConversion(DECIMAL_CONVERSION);
+      addLogicalTypeConversion(UUID_CONVERSION);
+      addLogicalTypeConversion(DATE_CONVERSION);
+      addLogicalTypeConversion(TIME_MICROS_CONVERSION);
+      addLogicalTypeConversion(TIMESTAMP_MICROS_CONVERSION);
+      // NOTE: Those are not supported in Avro 1.8.2
+      // TODO re-enable upon upgrading to 1.10
```

Comment:
makes sense.. maybe also create a tracking jira for avro upgrade if there isn't alrady one.. there are some GH issues tied to the upgrade too.